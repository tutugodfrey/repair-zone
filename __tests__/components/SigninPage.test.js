import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ConnectedSigninPage, { SigninPage, mapStateToProps }
  from '../../client/src/components/SigninPage.jsx';

let wrapper;
let mockStore = configureStore();
let store = mockStore();
describe('<SignupPage /> test', () => {
  describe('unconnected signup page', () => {
    beforeEach(() => {
      const dispatch = jest.fn();
      wrapper = shallow(
        <SigninPage
          dispatch={dispatch}
        />
      );
    });
    test('should map the snapshot of the component correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    })
  });

  describe('connected signup page', () => {
    let wrapper;
    beforeEach(() => {
      const dispatch = jest.fn();
      wrapper = mount(
        <Provider store={store}>
          <ConnectedSigninPage
            dispatch={dispatch}
          />
        </Provider>
      );
    });
    test('should render the page with a signup button', () => {
      const signupBtn = wrapper.find('button');
      expect(signupBtn.length).toBe(2);
    });
    it('should call connected functions', () => {
    })
  });
});
