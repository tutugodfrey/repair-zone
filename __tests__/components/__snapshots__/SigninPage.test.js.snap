// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupPage /> test unconnected signup page should map the snapshot of the component correctly 1`] = `
<div>
  <div
    className="py-3 pl-5 bg-success"
    id="logo-div-2"
  >
    <a
      className="text-white font-weight-bold"
      href="#"
      onClick={[Function]}
    >
      Repair-Zone
    </a>
  </div>
  <Connect(Modal) />
  <div
    className="row"
  >
    <div
      className="card py-2 border-success px-5 col-10 offset-col-1 col-sm-6 col-md-4 offset-md-4 offset-sm-3"
      id="login-div"
    >
      <div
        className="card-body"
      >
        <form
          action=""
        >
          <h3
            className="card-title"
          >
            Sign In
          </h3>
          <FormInput
            divClass="my-2"
            divId=""
            inputClass="ml-1 border-success form-control field-valid required"
            inputId="signin-2-username"
            inputName="username"
            inputType="text"
            labelId=""
            labelValue="Username"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            placeholder="username"
            requiredField={true}
            spanClass="required text-danger"
            validationClass="text-danger"
          />
          <FormInput
            divClass="my-2"
            divId=""
            inputClass="ml-1 border-success form-control field-valid required"
            inputId="signin-2-password"
            inputName="password"
            inputType="password"
            labelId=""
            labelValue="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            placeholder="password"
            requiredField={true}
            spanClass="required text-danger"
            validationClass="text-danger"
          />
          <Button
            buttonClass="ml-1 my-2 px-5 py-2 btn btn-sm bg-success text-white"
            buttonId=""
            buttonName="log-In"
            onClick={[Function]}
          />
          <p>
            You don't have an account? 
            <br />
            <Link
              href="#"
              hrefClass="card-link"
              linkText="Sign Up"
              onClick={[Function]}
            />
            Â here
          </p>
        </form>
      </div>
    </div>
  </div>
</div>
`;
